app/services/request_reference.rb

```diff
@@ -1,4 +1,18 @@
 class RequestReference
+  include ActiveModel::Model
+  attr_accessor :reference
+
+  validate :reference_complete_information, if: :new_reference_flow?
+
+  def send_request
+    return unless valid?
+
+    # Backwards compatibility between old and new references
+    # because I need to validate the @reference object
```

Comment:
i don't understand this comment

Reply:
Yes, the comment is not great but I decided to not touch on the behaviour of old references, maybe I should change this comment to "This method save is used in the new references and for backwards compatibility with old references I need to call the #call method". Is it better?

Reply:
Done! ♻️